sky/packages/sky/lib/widgets/basic.dart

```diff
@@ -298,8 +298,9 @@ class Viewport extends OneChildRenderObjectWrapper {
 
   void syncRenderObject(Viewport old) {
     super.syncRenderObject(old);
-    renderObject.scrollOffset = scrollOffset;
+    // Order dependency: RenderObject validates scrollOffset based on scrollDirection.
     renderObject.scrollDirection = scrollDirection;
+    renderObject.scrollOffset = scrollOffset;
```

Comment:
To satisfy my OCD, could you change the order of these fields and arguments throughout this class so they're consistent throughout?